packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m

```diff
@@ -407,317 +408,222 @@ - (GMSCameraPosition *)cameraPosition {
-  return [FLTGoogleMapJsonConversions toLocation:data];
-}
-
-static int ToInt(NSNumber *data) { return [FLTGoogleMapJsonConversions toInt:data]; }
-
-static BOOL ToBool(NSNumber *data) { return [FLTGoogleMapJsonConversions toBool:data]; }
-
-static CGPoint ToPoint(NSArray *data) { return [FLTGoogleMapJsonConversions toPoint:data]; }
-
-static GMSCameraPosition *ToCameraPosition(NSDictionary *data) {
-  return [GMSCameraPosition cameraWithTarget:ToLocation(data[@"target"])
-                                        zoom:ToFloat(data[@"zoom"])
-                                     bearing:ToDouble(data[@"bearing"])
-                                viewingAngle:ToDouble(data[@"tilt"])];
-}
-
-static GMSCameraPosition *ToOptionalCameraPosition(NSDictionary *json) {
-  return json ? ToCameraPosition(json) : nil;
-}
-
-static CGPoint ToCGPoint(NSDictionary *json) {
-  double x = ToDouble(json[@"x"]);
-  double y = ToDouble(json[@"y"]);
-  return CGPointMake(x, y);
-}
-
-static GMSCoordinateBounds *ToBounds(NSArray *data) {
-  return [[GMSCoordinateBounds alloc] initWithCoordinate:ToLocation(data[0])
-                                              coordinate:ToLocation(data[1])];
-}
-
-static GMSCoordinateBounds *ToOptionalBounds(NSArray *data) {
-  return (data[0] == [NSNull null]) ? nil : ToBounds(data[0]);
-}
-
-static GMSMapViewType ToMapViewType(NSNumber *json) {
-  int value = ToInt(json);
-  return (GMSMapViewType)(value == 0 ? 5 : value);
-}
-
-static GMSCameraUpdate *ToCameraUpdate(NSArray *data) {
-  NSString *update = data[0];
-  if ([update isEqualToString:@"newCameraPosition"]) {
-    return [GMSCameraUpdate setCamera:ToCameraPosition(data[1])];
-  } else if ([update isEqualToString:@"newLatLng"]) {
-    return [GMSCameraUpdate setTarget:ToLocation(data[1])];
-  } else if ([update isEqualToString:@"newLatLngBounds"]) {
-    return [GMSCameraUpdate fitBounds:ToBounds(data[1]) withPadding:ToDouble(data[2])];
-  } else if ([update isEqualToString:@"newLatLngZoom"]) {
-    return [GMSCameraUpdate setTarget:ToLocation(data[1]) zoom:ToFloat(data[2])];
-  } else if ([update isEqualToString:@"scrollBy"]) {
-    return [GMSCameraUpdate scrollByX:ToDouble(data[1]) Y:ToDouble(data[2])];
-  } else if ([update isEqualToString:@"zoomBy"]) {
-    if (data.count == 2) {
-      return [GMSCameraUpdate zoomBy:ToFloat(data[1])];
-    } else {
-      return [GMSCameraUpdate zoomBy:ToFloat(data[1]) atPoint:ToPoint(data[2])];
-    }
-  } else if ([update isEqualToString:@"zoomIn"]) {
-    return [GMSCameraUpdate zoomIn];
-  } else if ([update isEqualToString:@"zoomOut"]) {
-    return [GMSCameraUpdate zoomOut];
-  } else if ([update isEqualToString:@"zoomTo"]) {
-    return [GMSCameraUpdate zoomTo:ToFloat(data[1])];
-  }
-  return nil;
-}
-
-static void InterpretMapOptions(NSDictionary *data, id<FLTGoogleMapOptionsSink> sink) {
++ (void)interpretMapOptions:(NSDictionary *)data sink:(id<FLTGoogleMapOptionsSink>)sink {
```

Comment:
Same comment and pattern as `FLTGoogleMapCircleOptionsSink` and `FLTGoogleMapTileOverlayOptionsSink`, does `FLTGoogleMapOptionsSink` need to exist as a protocol?  It seems like they should just be properties on `FLTGoogleMapController` since that's the only place it's used.